var/spack/repos/builtin/packages/py-aiohttp/package.py

```diff
@@ -13,18 +13,26 @@ class PyAiohttp(PythonPackage):
     homepage = "https://github.com/aio-libs/aiohttp"
     url      = "https://github.com/aio-libs/aiohttp/archive/v3.6.2.tar.gz"
 
+    version('3.8.1', sha256='45210606552ed4f8992e17ffa3273e9ad2815c8c3ca00832a57b3a62735b1fc7')
+    version('3.8.0', sha256='0587288bf063cc0433542ab17674af11cb3639f4cb5ddf89420501ca16565e75')
     version('3.7.4', sha256='d90ace66b55747e49531b13bf0a9c7ae9d1e14f315a56ea6b54c0dc1b6facd6e')
     version('3.6.2', sha256='4ad2d8393843ea0c7b50a93bf04fb1cf20b93c0a5e958de128efcd2e5f8adf80')
 
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools@46.4:', type='build')
+    depends_on('py-charset-normalizer@2.0:2', type=('build', 'run'), when='@3.8.0:')
     depends_on('python@3.5.3:', type=('build', 'run'))
     depends_on('python@3.6:', type=('build', 'run'), when='@3.7:')
     depends_on('py-attrs@17.3.0:', type=('build', 'run'))
     depends_on('py-chardet@2.0:3', type=('build', 'run'))
     depends_on('py-multidict@4.5:4', type=('build', 'run'), when='@:3.6.2')
     depends_on('py-multidict@4.5:6', type=('build', 'run'), when='@3.6.3:')
-    depends_on('py-async-timeout@3.0:3', type=('build', 'run'))
+    depends_on('py-async-timeout@3.0:3', type=('build', 'run'), when='@:3.7.4')
+    depends_on('py-async-timeout@4.0:4', type=('build', 'run'), when='@3.8.0:')
+    depends_on('py-asynctest@0.13.0', type=('build', 'run'), when='@3.8.0: ^python@:3.7')
     depends_on('py-yarl@1.0:1', type=('build', 'run'))
     depends_on('py-idna-ssl@1.0:', type=('build', 'run'), when='^python@:3.6')
+    depends_on('py-typing-extensions@3.7.4:', type=('build', 'run'), when='@3.8.1: ^python@:3.7')
     depends_on('py-typing-extensions@3.6.5:', type=('build', 'run'), when='^python@:3.6')
     depends_on('py-typing-extensions@3.6.5:', type=('build', 'run'), when='@3.7.1:')
```

Comment:
These two should probably be guarded by version checks. Otherwise, this means that any version above 3.7.1 (including 3.8.1) OR any version built with Python 3.6 will need this dep.

Reply:
To double check but I think it's now corrected.